Core/Object Arts/Dolphin/MVP/Presenters/List/ListBox.cls

```diff
@@ -443,13 +453,29 @@ showDropHighlight: anItem
 
 state
 	"Private - Answer a MessageSequence which, when replayed, will restore the receiver 
-	to its current state. Add the horizonatl extent aspect since it is not otherwise cached in an
-	instance variable"
-
-	^super state
-		add: (MessageSend receiver: self selector: #horizontalExtent: argument: self horizontalExtent);
-		yourself
-!
+	to its current state."
+
+	| messages |
+	messages := super state.
+	"Add the horizontal extent aspect since it is not otherwise cached in an instance variable."
+	messages add: (MessageSend
+				receiver: self
+				selector: #horizontalExtent:
+				argument: self horizontalExtent).
+	"The caret and anchor only need to be restored separately in multi-select mode--
+	in single-select mode they are always equal to the selected item."
+	self isMultiSelect ifTrue: 
+			[self anchorIndex = 0 ifFalse: 
```

Comment:
Given that `anchorIndex` and `caretIndex` are computed by sending a message to the control (which will call back into Dolphin, before being passed back to the control's own window procedure), it would be worthwhile storing the returned values into temps to avoid having to compute each twice.